cmd/config/internal/commands/cmdxargs_test.go

```diff
@@ -85,7 +85,7 @@ func TestXArgs_flags(t *testing.T) {
 	if !assert.NoError(t, c.Command.Execute()) {
 		t.FailNow()
 	}
-	assert.Equal(t, strings.TrimSpace(`--a=b --c=d --e=f 1 3 2 4`), strings.TrimSpace(out.String()))
+	assert.Contains(t, strings.TrimSpace(out.String()), strings.TrimSpace(`--a=b --c=d --e=f 1 3 2 4`))
```

Comment:
Does this still need to be changed to `Contains` now?

Reply:
No it doesn't! I reverted this line, thanks for the catch

Reply:
Can all the similar changes also be reverted to the stricter versions of the assertions?

Reply:
The similar assertions are in `sink_test` and `source_test`. Currently  in this PR `sink` and `source` are using the `Deprecated` field of the cobra.Command, which sends the deprecation message to stdout along with the rest of the output, so the similar changes cannot be reverted as is.

The alternative would be to take the same approach in `sink` and source` as done in `wrap` and `xargs` - instead of using the `Deprecated` field of the cobra.Command, we would have to similarly introduce a `preRunE` function to output a deprecation message.

I personally prefer to leave it as is, the `Deprecated` field feels more standard and reliable than the alternative approach of using `preRunE`, which is really just a workaround to get around the fact that `xargs` and `wrap` use stdout as input to the next filter in the pipeline. WDYT?

Reply:
Unless I'm misunderstanding how they work, all these commands emit structured output for consumption and will break if we start printing deprecations to stdout. In the following:

`kustomize fn source DIR/ | your-function | kustomize fn sink DIR/`

We don't want `your-function` to receive the deprecation, or the directory to include it; in both cases that will make the yaml in question invalid.

I'm really surprised Cobra sends deprecation messages to stdout by default. My favourite explanation of when to use which stream is not online anymore, but you can still find it archived [here](https://web.archive.org/web/20190816165026/https://www.jstorimer.com/blogs/workingwithcode/7766119-when-to-use-stderr-instead-of-stdout). Notably:

<img width="733" alt="image" src="https://user-images.githubusercontent.com/4789493/133541976-db829d3a-37b8-4019-9289-851666645368.png">

Reply:
> We don't want your-function to receive the deprecation, or the directory to include it; in both cases that will make the yaml in question invalid.

You're right. Will do.

Reply:
done! ready for another look